python/pyarrow/hdfs.py

```diff
@@ -146,8 +146,12 @@ def _derive_hadoop_classpath():
     xargs_echo = subprocess.Popen(('xargs', 'echo'),
                                   stdin=find.stdout,
                                   stdout=subprocess.PIPE)
-    return subprocess.check_output(('tr', "' '", "':'"),
-                                   stdin=xargs_echo.stdout)
+    classpath_jars = subprocess.check_output(('tr', "' '", "':'"),
+                                             stdin=xargs_echo.stdout)
+    classpath_conf = os.environ["HADOOP_CONF_DIR"] \
+        if "HADOOP_CONF_DIR" in os.environ \
+        else os.environ['HADOOP_HOME'] + "/etc/hadoop"
+    return (classpath_conf + ":").encode('utf-8') + classpath_jars
```

Comment:
this looks a little strange because you are append class path jars found in os.environ['HADOOP_HOME'] to possible os.environ['HADOOP_CONF_DIR'] is that what is intended?g

Reply:
> this looks a little strange because you are append class path jars found in os.environ['HADOOP_HOME'] to possible os.environ['HADOOP_CONF_DIR'] is that what is intended?g

when pyarrow hdfs use libhdfs, hadoop config dir must be in classpath,but the _derive_hadoop_classpath only add jar files into classpath,hadoo config dir is not in the classpath,and the libhdfs can not locate config  files (hdfs-site.xml,core-site.xml etc )correctly

Reply:
HADOOP_CONF_DIR  let user customize hadoop config directory, eg /etc/hadoop 
if HADOOP_CONF_DIR is not set,then $HADOOP_HOME/etc/home as the default value